src/main/com/fulcrologic/fulcro/application.cljc

```diff
@@ -393,7 +395,8 @@
                                        ::root-factory root-factory
                                        ::root-class root)
                                      (update-shared! app)
-                                     (indexing/index-root! app)
+                                     (util/check-query-valid (comp/get-query root) comp/component-name)
```

Comment:
Same as the other comment. `get-query` on initial mount is probably just the static query declared on components, but on remount it needs to be the current dynamically bound query, which requires you pass the state map to `get-query`.